src/GanttChart.hs

```diff
@@ -0,0 +1,136 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module GanttChart where
+
+import Control.Lens.TH (makeLenses)
+import Parser
+import Text.Megaparsec as M
+import Text.Megaparsec.Char (alphaNumChar, char, string)
+
+data GanttChartGraph = GanttChartGraph
+  { _ganttChartTitle :: GanttChartTitle,
+    _dateFormat :: DateFormat,
+    _axisFormat :: AxisFormat,
+    _section :: [Section]
+  }
+  deriving (Eq, Show, Generic)
+
+data Section = Section
+  { _sectionTitle :: SectionTitle,
+    _tasks :: [Task]
+  }
+  deriving (Eq, Show, Generic)
+
+data Task = Task
+  { _taskName :: TaskName,
+    _taskState :: Maybe TaskState
+  }
+  deriving (Eq, Show, Generic)
+
+newtype GanttChartTitle = GanttChartTitle {unGanttChartTitle :: Text} deriving (Eq, Show, Generic)
+
+newtype DateFormat = DateFormat {unDateFormat :: Text} deriving (Eq, Show, Generic)
+
+newtype AxisFormat = AxisFormat {unAxisFormat :: Text} deriving (Eq, Show, Generic)
+
+newtype SectionTitle = SectionTitle {unSectionTile :: Text} deriving (Eq, Show, Generic)
+
+newtype TaskName = TaskName {unTaskName :: Text} deriving (Eq, Show, Generic)
+
+data TaskState = Done | Active | Crit | After deriving (Eq, Show, Generic)
+
+makeLenses ''GanttChartGraph
+
+makeLenses ''GanttChartTitle
+
+makeLenses ''DateFormat
+
+makeLenses ''AxisFormat
+
+makeLenses ''Section
+
+makeLenses ''SectionTitle
+
+makeLenses ''Task
+
+makeLenses ''TaskName
+
+makeLenses ''TaskState
+
+instance IsString GanttChartGraph where
+  fromString s = GanttChartGraph (fromString s) "" "" []
+
+instance IsString Section where
+  fromString s = Section (fromString s) []
+
+instance IsString GanttChartTitle where
+  fromString s = GanttChartTitle (fromString s)
+
+instance IsString DateFormat where
+  fromString s = DateFormat (fromString s)
+
+instance IsString AxisFormat where
+  fromString s = AxisFormat (fromString s)
+
+instance IsString SectionTitle where
+  fromString s = SectionTitle (fromString s)
+
+instance IsString Task where
+  fromString s = Task (fromString s) Nothing
+
+instance IsString TaskName where
+  fromString s = TaskName (fromString s)
+
+pGanttChartTitle :: Parser GanttChartTitle
+pGanttChartTitle = do
+  void $ lexeme $ string "title"
+  lexeme (fromString <$> M.some (alphaNumChar <|> char ' ' <|> char '?' <|> char '!' <|> char '\'' <|> char ',' <|> char ':' <|> char '%'))
```

Comment:
Perhaps factor out `alphaNumChar <|> char ' ' <|> char '?' <|> char '!' <|> char '\'' <|> char ',' <|> char ':' <|> char '%')` as its own parser function, so it can be reused in several places.